cfgmgr/portmgr.cpp

```diff
@@ -22,7 +23,40 @@ PortMgr::PortMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, c
 {
 }
 
-bool PortMgr::setPortMtu(const string &alias, const string &mtu)
+bool PortMgr::isLagMember(const std::string &alias, std::unordered_map<std::string, int> &lagMembers)
+{
+    /* Lag members are lazily loaded on the first call to isLagMember and cached
+     * within a variable inside of doTask() for future calls.
+     */
+    if (lagMembers.empty())
+    {
+        vector<string> keys;
+        m_cfgLagMemberTable.getKeys(keys);
```

Comment:
This will not handle if lag member gets added after bootup.

Reply:
doTask() isn't a long-running process and the cache is local to doTask() and not global, so it doesn't survive past a single call to doTask().  The only thing the cache really helps with is on boot where a large number of ports are being brought up so it will have more than 1 task to iterate across in doTask() ... I'd assume in most instance doTask() will only have a single task outside of bootup (I could be wrong here, I haven't tried to trace it).

Reply:
doTask is called for every update from config_db. So the cache would be outdated if someone creates a portchannel using CLI.

Reply:
No, that's incorrect. 

The cache is a local variable to doTask().  Cache is destroyed at the end of each doTask call because the cache variable goes out of scope at the end of the function call.

So subsequent calls would create a new temporary cache. its not persistent from call to call.

Unless I'm seriously missing something here, but it looks like doTask() is a short-lived process.  It loops through an iterator processing events and exits ... just to be called again ... just looking at the main() loop for portmgrd.cpp its not expected to block for the life of the daemon or anything like that.

Reply:
This could have performance issues with scaled up lag member and each time traversing through the list. I think for your specific issue, we could follow how this is done currently in sonic-mgmt tests as a work-around, to configure the lag

Reply:
what do you mean by "scaled up lag member"? 

This code is only called if the port is NOT in the PORT_TABLE.  I'm not aware of any instances where ports are added to PORT_TABLE after startup.  

During startup, it is expected that doTask() will not be invoked per port, but rather receive events for multiple (likely ***all***) ports at once, so I optimized for that case by adding a hashtable cache each port won't have to pull the list for the db, its pulled at most once per doTask() call.

Therefore, I expect the overhead of this patch to be exactly 1 db query per startup.  Then a very cheap hashtable lookup once per port.